File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        row_set, col_set = set(), set()
        m, n = len(matrix), len(matrix[0])

        for i in range(m):
            for j in range(n):
                if matrix[i][j] == 0:
                    row_set.add(i)
                    col_set.add(j)
        
        for row in row_set:
            for j in range(n):
                matrix[row][j] = 0

        for col in col_set:
            for i in range(m):
                matrix[i][col] = 0

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {

        Set<Integer> row_set = new HashSet<Integer>();
        Set<Integer> col_set = new HashSet<Integer>();        

        for(int i=0; i<matrix.length; i++){
            for(int j=0; j<matrix[i].length; j++){
                if(matrix[i][j] == 0){
                    row_set.add(i);
                    col_set.add(j);
                }
            }
        }

        for(int num : row_set){
            for(int j=0; j<matrix[0].length; j++){
                matrix[num][j] = 0;
            }
        }

        for(int num : col_set){
            for(int i=0; i<matrix.length; i++){
                matrix[i][num] = 0;
            }
        }        
    }
}